madanalysis/layout/plotflow.py

```diff
@@ -110,865 +123,1162 @@ def InitializeHistoFrequency(self,ihisto):
-
-            output_files=[]
-            for iout in range(0,len(output_paths)):
-                output_files.append('../../'+output_paths[iout].split('/')[-2]+'/'+\
-                                    output_paths[iout].split('/')[-1]+"/selection_"+str(irelhisto)+"."+\
-                                    ReportFormatType.convert2filetype(modes[iout]))
-
-            for iset in range(0,len(self.detail)):
-            # Appending histo
-                histos.append(self.detail[iset][irelhisto])
-#               if mode==2:
-                scales.append(self.detail[iset][irelhisto].scale)
-#               else:
-#                   scales.append(1)
-
-            logging.getLogger('MA5').debug('Producing file '+filenameC+' ...')
-            self.DrawROOT(histos,scales,self.main.selection[iabshisto],\
-                          irelhisto,filenameC,output_files)
-
-            logging.getLogger('MA5').debug('Producing file '+filenamePy+' ...')
-            self.DrawMATPLOTLIB\
-                         (histos,scales,self.main.selection[iabshisto],\
-                          irelhisto,filenamePy,output_files)
-
-            irelhisto+=1
-
+            filenameC = histo_path + "/selection_" + str(irelhisto) + ".C"
+            filenamePy = histo_path + "/selection_" + str(irelhisto) + ".py"
+
+            output_files = []
+            for iout, outp in enumerate(output_paths):
+                output_files.append(
+                    "../../"
+                    + outp.split("/")[-2]
+                    + "/"
+                    + outp.split("/")[-1]
+                    + "/selection_"
+                    + str(irelhisto)
+                    + "."
+                    + ReportFormatType.convert2filetype(modes[iout])
+                )
+
+            for detail in self.detail:
+                # Appending histo
+                datasets.append(detail.dataset)
+                histos.append(detail[irelhisto])
+                scales.append(detail[irelhisto].scale)
+
+            logging.getLogger("MA5").debug("Producing file " + filenameC + " ...")
+            if self.main.archi_info.has_root:
+                self.DrawROOT(histos, scales, select, irelhisto, filenameC, output_files)
+
+            logging.getLogger("MA5").debug("Producing file " + filenamePy + " ...")
+            self.DrawMATPLOTLIB(
+                histos, scales, datasets, select, irelhisto, filenamePy, output_files
+            )
+
+            irelhisto += 1
 
         # Save ROOT files
-        for ind in range(0,irelhisto):
-            ListROOTplots.append(histo_path+'/selection_'+str(ind))
-            
-        return True
+        for ind in range(0, irelhisto):
+            ListROOTplots.append(histo_path + "/selection_" + str(ind))
 
+        return True
 
-    def DrawROOT(self,histos,scales,ref,irelhisto,filenameC,outputnames):
+    def DrawROOT(self, histos, scales, ref, irelhisto, filenameC, outputnames):
```

Comment:
So there are no error bars in root, aren't there?

Reply:
to hell with root